### DISCUSSION TITLE
How to de/serialise sealed classes with objects

### DISCUSSION THREAD

**Discussion by jbruchanov** [pdvrieze/xmlutil#149] [Q&A]
I'm fighting little bit with the annotation here, anyone would be able to help ?

having following code
```
@Serializable
@XmlElement
@XmlSerialName("doc")
class Doc(
    @XmlElement
    @XmlSerialName("vars")
    val vars: List<DocVar>
)

@XmlElement
@Serializable
data class DocVar(
    @XmlElement
    @XmlSerialName("name")
    val name: String,

    @XmlElement
    @XmlSerialName("dataType")
    val dataType: DocVarType,
)


@Serializable
sealed class DocVarType {
    @Serializable
    @XmlSerialName("string")
    object String : DocVarType()
}

val obj = Doc(
    vars = listOf(
        DocVar(
            name = "testname",
            dataType = DocVarType.String
        )
    )
)
```            

what I'm trying to do 
```
<doc>
    <vars>
        <name>testname</name>
        <dataType>string</dataType>   <-- not sure how to do this
    </vars>
</doc>
```

but getting stuff like
```
<dataType type="DocVarType.String">
    <value/>
</dataType>

or directly <string />
```

**Comment by pdvrieze** [comment]
The problem is that (not specific to XML) objects are only serialized in their existence (an empty tag by default in XML - it is possible to switch to attributes in the policy), not with any members. A separate thing is that an object inside a class is not actually a member of the instance. The actual strings are a value, but never a type, so there is no annotation to write the string value as tag content. In this case it is probably easiest to use a custom serializer, or not use an object but an actual value.
